snowpack/src/rollup-plugins/rollup-plugin-node-process-polyfill.ts

```diff
@@ -34,17 +36,17 @@ function createProcessPolyfill(vars: EnvVarReplacements = {}) {
       [id]: vars[id] === true ? process.env[id] : vars[id],
     };
   }, {});
+  const polyfillPath = require.resolve('rollup-plugin-node-polyfills/polyfills/process-es6.js');
+  const polyfillCode = fs
+    .readFileSync(polyfillPath, 'utf-8')
+    .replace(/^export default .*$/gm, '')
```

Comment:
ah, I didn't realize how many replacements would be needed. It would be great to pin to the dependency like this so that we get automatic updates, but now the fear is that those updates break our replacers.

In that case, I'd recommend grabbing the polyfill code and owning it ourselves, so that there's no risk of breakage and no need to live-replace these file contents.

In a new file:
```
/*
(The MIT License)

Copyright (c) 2013 Roman Shtylman <shtylman@gmail.com>

Permission is hereby granted, free of charge, to any person obtaining
a copy of this software and associated documentation files (the
'Software'), to deal in the Software without restriction, including
without limitation the rights to use, copy, modify, merge, publish,
distribute, sublicense, and/or sell copies of the Software, and to
permit persons to whom the Software is furnished to do so, subject to
the following conditions:

The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
*/

export function generateProcessPolyfill(env) {
  return `
    COPY-PASTE 
    https://github.com/calvinmetcalf/node-process-es6/blob/master/browser.js
    INTO THIS TEMPLATE STRING
    MAKE ANY EDITS THAT WE NEED, DIRECTLY
  `;
}
```

Then our rollup plugin can just do:

```
return {code: generateProcessPolyfill(env), moduleSideEffects: false}`;
```

Reply:
good advice.